update focus-ring content (#954)
hashicorp/design-system: website/docs/foundations/focus-ring/index.md

description: Focus rings display when interactive components including links and buttons are focused by the user. They are a vital tool in communicating interactivity.
---

## How to use this style

> The suggested way to apply a "focus-ring" style to an UI element is using the specific **design token** provided as CSS custom property.

Reviewer: ```suggestion
The suggested way to apply a “focus-ring” style to an UI element is using the specific **design token** provided as CSS custom property.
```

### Design tokens

You can use the `--token-focus-ring-action-box-shadow` [design token](./tokens) directly in your CSS definitions:

```css
.your-selector {
  [...your CSS declarations]
  &:focus,
  &:focus-visible {
    box-shadow: var(--token-focus-ring-action-box-shadow);
  }
}
```

!!! Critical

> *   The design token as CSS variable can be used **only** with a `box-shadow` property

Reviewer 2: ```suggestion
*   The design token as CSS variable can be used **only** with a `box-shadow` property.
```

> *   The border radius depends on the UI element to which is applied to, so it's up to you to apply the right `border-radius` (tip: consider to use the `inherit` value).

Reviewer 2: ```suggestion
*   The border radius depends on the UI element to which it is applied to, so it's up to you to apply the right `border-radius` (tip: consider to use the `inherit` value).
```

> *   The border radius depends on the UI element to which is applied to, so it's up to you to apply the right `border-radius` (tip: consider to use the `inherit` value).

Reviewer: ```suggestion
*   The border radius depends on the UI element to which it is applied, so it’s up to you to apply the correct `border-radius` (tip: consider using the `inherit` value).
```

!!!

### CSS helper classes

> We provide also a **CSS helper class** `.hds-focus-ring-box-shadow`, that is a wrapper of the design token. It's unlikely you can use it directly in a template because this style is connected to the "focused" pseudo-state of an element (more likely it would be used in composition with other classes).

Reviewer 2: ```suggestion
We provide a **CSS helper class** `.hds-focus-ring-box-shadow`, that is a wrapper of the design token. It's unlikely you can use it directly in a template because this style is connected to the "focused" pseudo-state of an element (more likely it would be used in composition with other classes).
```

To use this class you have to import the CSS file `[products|devdot]/css/helpers/focus-ring.css` from the `@hashicorp/design-system-tokens` package.

!!! Critical

> The border radius depends on the UI element to which is applied to, so it's up to you to apply the right `border-radius` (tip: consider to use the `inherit` value).

Reviewer 2: ```suggestion
The border radius depends on the UI element to which it is applied to, so it's up to you to apply the right `border-radius` (tip: consider to use the `inherit` value).
```

!!!

### Sass mixins

> We have also created two **Sass mixins** `hds-focus-ring-basic` and `hds-focus-ring-with-pseudo-element`, but they're mainly used for internal use (to the design system codebase). These mixins do more than just apply the focus style: they also take care of all the different way to declare the `:focus/:focus-visible` for different browsers.

Reviewer 2: ```suggestion
We have created two **Sass mixins** `hds-focus-ring-basic` and `hds-focus-ring-with-pseudo-element`, but they're mainly used for internal use (to the design system codebase). These mixins do more than just apply the focus style: they also take care of all the different way to declare the `:focus/:focus-visible` for different browsers.
```
